52/ttafsir/pomodoro.py

```diff
@@ -0,0 +1,39 @@
+from rich.console import Console
+from rich.live import Live
+from time import sleep
+from datetime import timedelta
+
+
+console = Console()
+
+
+class PomodoroTimer:
+    def __init__(self, start_timer=25, break_timer=5):
+        self._timer = start_timer
+        self._break = break_timer
+        self._countdown = timedelta(minutes=self._timer)
+
+    def run(self, prompt="TIMER"):
+        with Live(auto_refresh=False) as live:
+            while self._countdown:
+                console.print(f"{prompt}: {self._countdown}", end="\r")
+                self._countdown += timedelta(seconds=-1)
+                sleep(1)
+                live.refresh()
+
+    def restart(self, start_timer, prompt):
+        self._countdown = timedelta(minutes=start_timer)
+        self.run(prompt)
+
+
+timer_duration = int(input("Duration: ").strip())
```

Comment:
Put this into an `if __name__ == "__main__"` so it doesn't run if you would want to import this module.

Reply:
ah, good point. will do that.

Reply:
Thanks!